src/glamor/CSSPropertyOperations/index.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule CSSPropertyOperations
+ */
+
+import camelizeStyleName from 'fbjs/lib/camelizeStyleName'
+import dangerousStyleValue from './dangerousStyleValue'
+import hyphenateStyleName from 'fbjs/lib/hyphenateStyleName'
+import memoizeStringOnly from 'fbjs/lib/memoizeStringOnly'
+import warning from 'fbjs/lib/warning'
+
+export const processStyleName = memoizeStringOnly(hyphenateStyleName)
+
+if (process.env.NODE_ENV !== 'production') {
+  // 'msTransform' is correct, but the other prefixes should be capitalized
+  let badVendoredStyleNamePattern = /^(?:webkit|moz|o)[A-Z]/
+
+  // style values shouldn't contain a semicolon
+  let badStyleValueWithSemicolonPattern = /;\s*$/
+
+  let warnedStyleNames = {}
+  let warnedStyleValues = {}
+  let warnedForNaNValue = false
+
+  let warnHyphenatedStyleName = function (name, owner) {
+    if (warnedStyleNames.hasOwnProperty(name) && warnedStyleNames[name]) {
+      return
+    }
+
+    warnedStyleNames[name] = true
+    process.env.NODE_ENV !== 'production'
```

Comment:
I'd move all these `!isProd ? warning(...) : void 0` to a helper function -- probably a small wrapper around existing `warning`.